packages/appium/test/e2e/cli.e2e.spec.js

```diff
@@ -296,6 +298,7 @@ describe('CLI behavior', function () {
           ret.fake.installType.should.eql('npm');
           ret.fake.installSpec.should.eql('@appium/fake-driver');
           const list = await runList(['--installed']);
+          // @ts-expect-error
```

Comment:
there are a couple ways we could go about it.  The error in particular is that the `list` var is an object of type `ExtensionListData`, which is:

```js
/**
 * Return value of {@link ExtensionCommand#list}.
 * @typedef {Record<string, Partial<import('../extension/manifest').InternalData> & ExtensionMetadata>} ExtensionListData
 */
```

`InternalData` isn't the problem, but `ExtensionMetadata` is:

```js
/**
 * Extra stuff about extensions; used indirectly by {@link ExtensionCommand#list}.
 *
 * @typedef {Object} ExtensionMetadata
 * @property {boolean} installed - If `true`, the extension is installed
 * @property {string|null} [updateVersion] - If the extension is installed, the version it can be updated to
 * @property {string|null} [unsafeUpdateVersion] - Same as above, but a major version bump
 * @property {boolean} [upToDate] - If the extension is installed and the latest
 */
```

This object requires the `installed` property; we cannot safely remove the property from the object via `delete`, which causes the error.

One way to fix this would just be to clone the `list` object w/o the `installed` prop, using something like `_.omit`.  That's reasonable, but it's perhaps more obvious just to remove the value, as the test is currently doing. We'd encounter a similar error if we did the opposite, e.g., add the `installed` prop to the `ret` object (which is type `ExtRecord<'driver'>`), since the property is unexpected.  _shrug_

Reply:
We could also make `installed` optional.  But that isn't quite true, because there's no case outside of test code where `ExtensionMetadata` does not have an `installed` prop.